src/documents/DataSourceAPI.md

```diff
@@ -0,0 +1,84 @@
+---
+title: DataSource API
+layout: default
+category: Infusion
+---
+
+The DataSource component provides the workflow for getting/setting (read/write) data from some data source. A data source may be an external database, cookie, in memory storage and etc. The specifics for accessing the data source must be provided in a concrete implementation. Additionally the payload of a request can be processed through an encoding step and other transformations during the get and set workflows.
+
+## Grades and Linkage
+
+There are two main grades `fluid.dataSource` and `fluid.dataSource.writable`. `fluid.dataSource` contains the base configuration and includes configuration for getting (reading) in from a data source. `fluid.dataSource.writable` adds in the configuration for setting (writing) to a data source. If you need to read and write to a data source, you'll need to add concrete implementations for both of these grades and make use of [grade linkage](IoCAPI.md#fluidmakegradelinkagelinkagename-inputnames-outputnames) to apply the concrete writable grade to the datasource configuration. In this way one would only need to add the `fluid.dataSource.writable` grade to gain the configuration from the concrete writable grade.
+
+```javascript
+fluid.defaults("my.dataSource", {
+    gradeNames: ["fluid.dataSource"]
+    // add grade specific configuration
+});
+
+fluid.defaults("my.dataSource.writable", {
+    gradeNames: ["fluid.dataSource.writable"]
+    // add grade specific configuration
+});
+
+fluid.makeGradeLinkage("my.dataSource.linkage", ["fluid.dataSource.writable", "my.dataSource"], "my.dataSource.writable");
+```
+
+## Encoding
+
+During the `get` (read) and `set` (write) workflows, listeners are provided as handles for encoding and transforming the payload. For example serializing and deserializing the payload on it's way to and from a server. The encoding is implemented by a supplied encoding component which includes `parse` and `render` methods. For `get` the data is run through `parse`. For `set`, the data is run through `render` when being sent to the server, and any returned values are sent through `parse`.
```

Comment:
Link at this point to the docs for fluid.promise.fireTransformEvent since it will not be clear in what sense these things are really "listeners". 
Typo it's -> its